CMakeLists.txt

```diff
@@ -98,23 +98,26 @@ if(BUILD_NETCDF OR BUILD_PNG)
     CMAKE_ARGS
     -DCMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}
     -DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}
+    -DCMAKE_C_FLAGS=-fPIC
     -DCMAKE_BUILD_TYPE=RELEASE
     SOURCE_DIR ${PROJECT_SOURCE_DIR}/zlib
     INSTALL_DIR ${CMAKE_INSTALL_PREFIX}
     )
+
+  ExternalProject_Add_Step(zlib remove_shared_lib
```

Comment:
This is to forcefully remove the shared library after the build? In the "old way" to build zlib (using configure), you could disable the building of the shared library. Can we do that either with cmake or by also reverting to configure, make and make install?

Reply:
zlib doesn't have an option to disable shared, either in CMake or configure. 

See

https://github.com/madler/zlib/issues/359

Reply:
You need to do this:
```
heinzeller-lt:zlib-1.2.11 dom.heinzeller$ ./configure --static
Checking for gcc...
Building static library libz.a version 1.2.11 with gcc.
Checking for size_t... Yes.
Checking for off64_t... No.
Checking for fseeko... Yes.
Checking for strerror... Yes.
Checking for unistd.h... Yes.
Checking for stdarg.h... Yes.
Checking whether to use vs[n]printf() or s[n]printf()... using vs[n]printf().
Checking for vsnprintf() in stdio.h... Yes.
Checking for return value of vsnprintf()... Yes.
Checking for attribute(visibility) support... Yes.
heinzeller-lt:zlib-1.2.11 dom.heinzeller$ make
gcc -O3 -DHAVE_HIDDEN -I. -c -o example.o test/example.c
gcc -O3 -DHAVE_HIDDEN  -c -o adler32.o adler32.c
gcc -O3 -DHAVE_HIDDEN  -c -o crc32.o crc32.c
gcc -O3 -DHAVE_HIDDEN  -c -o deflate.o deflate.c
gcc -O3 -DHAVE_HIDDEN  -c -o infback.o infback.c
gcc -O3 -DHAVE_HIDDEN  -c -o inffast.o inffast.c
gcc -O3 -DHAVE_HIDDEN  -c -o inflate.o inflate.c
gcc -O3 -DHAVE_HIDDEN  -c -o inftrees.o inftrees.c
gcc -O3 -DHAVE_HIDDEN  -c -o trees.o trees.c
gcc -O3 -DHAVE_HIDDEN  -c -o zutil.o zutil.c
gcc -O3 -DHAVE_HIDDEN  -c -o compress.o compress.c
gcc -O3 -DHAVE_HIDDEN  -c -o uncompr.o uncompr.c
gcc -O3 -DHAVE_HIDDEN  -c -o gzclose.o gzclose.c
gcc -O3 -DHAVE_HIDDEN  -c -o gzlib.o gzlib.c
gcc -O3 -DHAVE_HIDDEN  -c -o gzread.o gzread.c
gcc -O3 -DHAVE_HIDDEN  -c -o gzwrite.o gzwrite.c
libtool -o libz.a adler32.o crc32.o deflate.o infback.o inffast.o inflate.o inftrees.o trees.o zutil.o compress.o uncompr.o gzclose.o gzlib.o gzread.o gzwrite.o
gcc -O3 -DHAVE_HIDDEN -o example example.o -L. libz.a
gcc -O3 -DHAVE_HIDDEN -I. -c -o minigzip.o test/minigzip.c
gcc -O3 -DHAVE_HIDDEN -o minigzip minigzip.o -L. libz.a
```